src/cached_digest.ml

```diff
@@ -27,14 +29,49 @@ let cache = lazy (
   | None ->
     { checked_key = 0
     ; table = Path.Table.create 1024
+    ; max_timestamp = 0.
     }
   | Some cache ->
     cache.checked_key <- cache.checked_key + 1;
     cache)
 
+let get_current_filesystem_time () =
+  let special_path = Path.relative Path.build_dir ".filesystem-clock" in
+  let modify_file () =
+    let contents =
+      match Io.read_file special_path with
+      | exception Sys_error _ -> None
+      | exception Unix.Unix_error (ENOENT, _, _) -> None
+      | v -> match Int.of_string (String.trim v) with
+        | None -> None
+        | Some v -> Some v
+    in
+    let new_contents =
+      match contents with
+      | None -> "0"
+      | Some i -> Int.to_string (i + 1)
+    in
+    Io.write_lines special_path [new_contents]
+  in
+  modify_file ();
+  (Path.stat special_path).st_mtime
+
+let delete_very_recent_entries () =
+  let cache = Lazy.force cache in
+  let now = get_current_filesystem_time () in
+  match Float.compare cache.max_timestamp now with
+  | Lt -> ()
+  | Eq | Gt ->
+    Path.Table.filteri_inplace cache.table ~f:(fun ~key:_ ~data ->
+      match Float.compare data.timestamp now with
```

Comment:
Would be nice to just add the appropriate infix ops to `Float`. This is optional though.

Reply:
I find `compare` not any worse than infix ops. 
In fact I consistently read `Foo.(<) x` as "less than x" and get confused.